recipes/mpdecimal/2.5.x/conanfile.py

```diff
@@ -57,20 +53,16 @@ def validate(self):
         if is_msvc(self) and self.settings.arch not in ("x86", "x86_64"):
             raise ConanInvalidConfiguration(
                 f"{self.ref} currently does not supported {self.settings.arch}. Contributions are welcomed")
-        if self.options.cxx:
+        if self.options.cxx and Version(self.version) < "2.5.1":
             if self.options.shared and self.settings.os == "Windows":
                 raise ConanInvalidConfiguration(
                     "A shared libmpdec++ is not possible on Windows (due to non-exportable thread local storage)")
 
     def build_requirements(self):
-        if is_msvc(self):
-            self.tool_requires("automake/1.16.5")
-        else:
-            # required to support windows as a build machine
-            if self._settings_build.os == "Windows":
-                self.win_bash = True
-                if not self.conf.get("tools.microsoft.bash:path", check_type=str):
-                    self.tool_requires("msys2/cci.latest")
+        if not is_msvc(self) and self.settings_build.os == "Windows":
+            self.win_bash = True
```

Comment:
given that our CI only covers win+msvc, I suppose this is an untested branch now.
Is this to support mingw?

Chances are that we need to detect clang and check if it it needs to fall outside of this conditional as well

Reply:
I believe I tested this branch with Mingw and it worked. I can play around with it, but regardless of if it works or not this should still be a strict improvement, because the only change is just removing the automake dependency, which definitely was not used.

Reply:
I'm actually not very familiar with what `msys2` is used for in general. Is it just whenever we need access to a unix-style shell in a Windows build? So if we're calling commands like `ls`, `sed`, etc. (things that aren't available in Windows)?

Reply:
I tested mingw and clang. The former works, the latter I don't know how to get working, but I think that's beyond the scope of this PR. I'm certain that it didn't work before either.

Reply:
@jcar87 Can you take another look at this please?

Reply:
Thanks for testing this out in mingw!!